jobmon_client/src/jobmon/client/workflow.py

```diff
@@ -725,6 +727,22 @@ def _bind_tasks(
                 # get task resources id
                 self._set_original_task_resources(task)
 
+                serializable_resource_scales = copy.copy(task.resource_scales)
+                for resource, scaler in task.resource_scales.items():
+                    # We can't serialize a callable, so use the function name instead.
+                    if callable(scaler):
+                        serializable_resource_scales[resource] = getattr(
+                            scaler, "__name__", "Unknown Callable"
+                        )
+                    # We can't serialize an iterator, so take the relevant elements as a
+                    # list.
+                    elif isinstance(scaler, Iterator):
+                        serializable_resource_scales[resource] = list(
+                            itertools.islice(
+                                copy.deepcopy(scaler), task.max_attempts - 1
```

Comment:
Per my comment above, seems like this might be a duplicate deepcopy that's not needed

Reply:
Weirdly islice operates inplace, so if you don't deepcopy here you actually change what gets returned by `next`
<img width="287" alt="Screenshot 2023-09-08 at 2 29 57 PM" src="https://github.com/ihmeuw-scicomp/jobmon/assets/5368504/47dcfeb1-2ce2-4f30-a82d-4fb3bf158142">

Reply:
Not sure how to reply to the other comment, so replying here! I think we need it three times: 

The first two times are anywhere it's passed into a `Task` object, so once in `array.py` around line 266 and once in `task_template.py` around line 613. These are to give each `Task` its own copy of the `Iterator`, so each `Task` can iterate through independently.

The third time is in the serialization step, because `islice` will actually change the `next` value of the `Iterator`

Reply:
Noted! Guessing `islice` calls `next` under the hood if the input is an iterator. Thanks for addressing